src/app/ngrx/effects/castle-downloads.effects.ts

```diff
@@ -0,0 +1,241 @@
+import { Injectable } from '@angular/core';
+import { catchError, filter, map, mergeMap, switchMap } from 'rxjs/operators';
+import { of, Observable } from 'rxjs';
+import { Actions, Effect, ofType } from '@ngrx/effects';
+import { Action, Store, select } from '@ngrx/store';
+
+import * as ACTIONS from '../actions';
+import { selectRouter } from '../reducers/selectors';
+import { CastleService } from '@app/core';
+import { Episode, RouterParams, METRICSTYPE_DOWNLOADS, METRICSTYPE_DROPDAY } from '../';
+import * as dateUtil from '@app/shared/util/date';
+
+@Injectable()
+export class CastleDownloadsEffects {
+  routerParams: RouterParams;
+
+  // DOWNLOADS: whenever an episode page is loaded that is the currently routed page,
+  // call the load actions for podcast and episode downloads
+  @Effect()
+  loadRoutedDownloads$ = this.actions$.pipe(
+    ofType(ACTIONS.ActionTypes.CASTLE_EPISODE_PAGE_SUCCESS),
+    filter((action: ACTIONS.CastleEpisodePageSuccessAction) => {
+      const { page, episodes } = action.payload;
+      return this.routerParams.metricsType === METRICSTYPE_DOWNLOADS &&
+        episodes && episodes.length && this.routerParams && this.routerParams.episodePage === page;
+    }),
+    map((action: ACTIONS.CastleEpisodePageSuccessAction) => action.payload),
+    mergeMap((payload: ACTIONS.CastleEpisodePageSuccessPayload) => {
+      const { episodes } = payload;
+      this.store.dispatch(new ACTIONS.CastlePodcastDownloadsLoadAction({
+        id: episodes[0].podcastId,
+        interval: this.routerParams.interval,
+        beginDate: this.routerParams.beginDate,
+        endDate: this.routerParams.endDate
+      }));
+      this.store.dispatch(new ACTIONS.CastlePodcastAllTimeDownloadsLoadAction({id: episodes[0].podcastId}));
+      return episodes.map((episode: Episode) => {
+        this.store.dispatch(new ACTIONS.CastleEpisodeAllTimeDownloadsLoadAction({
+          podcastId: episode.podcastId,
+          guid: episode.guid
+        }));
+        return new ACTIONS.CastleEpisodeDownloadsLoadAction({
+          podcastId: episode.podcastId,
+          page: episode.page,
+          guid: episode.guid,
+          interval: this.routerParams.interval,
+          beginDate: this.routerParams.beginDate,
+          endDate: this.routerParams.endDate
+        });
+      });
+    })
+  );
+
+  // DROPDAY: when a page 1 of episodes is loaded
```

Comment:
the here is a new bit that loads episode drop day data after the first page of episodes is loaded to be selected by default